python/ray/experimental/sgd/utils.py

```diff
@@ -121,3 +126,16 @@ def update(self, val, n=1):
         self.sum += val * n
         self.count += n
         self.avg = self.sum / self.count
+
+
+def check_for_failure(remote_values):
```

Comment:
What kinds of failures is this intended to check for? Dead processes? Or methods that raised exceptions? Currently this handles both, but that seems to combine two very different use cases.

If you're specifically checking for dead actors, what about explicitly catching `RayActorException` or something like that?

Reply:
This helper function should be documented.

Reply:
done